### PR TITLE
chore(ci): run experimental delta xds on full matrix

### PR DESCRIPTION
**Pull request by lukidzi**
### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -479,6 +488,16 @@ workflows:
...
               arch: [ amd64, arm64 ]
           requires: [ build, check ]
+      - e2e:
+          name: << matrix.target >>:<< matrix.arch >>-<< matrix.k8sVersion >>-delta-kds
+          matrix:
+            alias: test/e2e-delta-kds
+            parameters:
+              k8sVersion: [ *lastK8sVersion ]
+              target: [ test/e2e-multizone ]
+              arch: [ amd64, arm64 ]
```
Comment (slonka) [inline .circleci/config.yml:498]: do we need both archs?

File: test/framework/config.go [hunk 2]
```diff
@@ -150,6 +151,13 @@ func (c E2eConfig) AutoConfigure() error {
 		Config.CIDR = "fd00:fd00::/64"
 	}
 
+	if Config.KumaDeltaKDS {
+		Config.KumaCpConfig.Multizone.KubeZone1.Envs["KUMA_EXPERIMENTAL_KDS_DELTA_ENABLED"] = "true"
+		Config.KumaCpConfig.Multizone.KubeZone2.Envs["KUMA_EXPERIMENTAL_KDS_DELTA_ENABLED"] = "true"
+		Config.KumaCpConfig.Multizone.UniZone1.Envs["KUMA_EXPERIMENTAL_KDS_DELTA_ENABLED"] = "true"
+		Config.KumaCpConfig.Multizone.UniZone2.Envs["KUMA_EXPERIMENTAL_KDS_DELTA_ENABLED"] = "true"
+	}
+
```
Comment (slonka) [inline test/framework/config.go:160]: should we run a mixture of experimental and old KDS?

File: .circleci/config.yml (hunk 1 above)
Comment (lukidzi) [reply .circleci/config.yml:498]: We don't need it, can run just `amd64` so far.

File: test/framework/config.go (hunk 2 above)
Comment (lukidzi) [reply test/framework/config.go:160]: It sounds more like a different use case. I would like to run just the new KDS to ensure that everything works correctly in the delta KDS.

**Review by slonka — APPROVED** [review]
I think we can go ahead with this

File: .github/workflows/blackbox-tests.yaml [hunk 3]
```diff
@@ -1,42 +1,33 @@
...
-
-    - name: "Setup user to test user id port exclusion"
-      run: |
-        sudo groupadd -g 4200 dns_test_user_group
-        sudo useradd dns_test_user -u 4201 -g 4200 -m -s /bin/bash
-
-    - name: "Run blackbox tests"
-      run: |
-        sudo "PATH=$PATH" $(which ginkgo) run ./test/blackbox_tests/...
+      - uses: actions/checkout@v3
```
Comment (michaelbeaumont) [inline .github/workflows/blackbox-tests.yaml:10]: Are these just formatting changes? Why are they in this PR?

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]

File: .github/workflows/blackbox-tests.yaml (hunk 3 above)
Comment (lukidzi) [reply .github/workflows/blackbox-tests.yaml:10]: Some strange conflict when merging, Going to fix it.

File: test/framework/config.go [hunk 4]
```diff
@@ -258,6 +266,7 @@ var defaultConf = E2eConfig{
 	ZoneIngressApp:       "kuma-ingress",
 	UniversalE2ELogsPath: path.Join(os.TempDir(), "e2e"),
 	CleanupLogsOnSuccess: false,
+	KumaDeltaKDS:         false,
```
Comment (michaelbeaumont) [inline test/framework/config.go:269]: this is automatically set by the env var?

File: test/framework/config.go [hunk 5]
```diff
@@ -150,6 +151,13 @@ func (c E2eConfig) AutoConfigure() error {
 		Config.CIDR = "fd00:fd00::/64"
 	}
 
+	if Config.KumaDeltaKDS {
+		Config.KumaCpConfig.Multizone.KubeZone1.Envs["KUMA_EXPERIMENTAL_KDS_DELTA_ENABLED"] = "true"
+		Config.KumaCpConfig.Multizone.KubeZone2.Envs["KUMA_EXPERIMENTAL_KDS_DELTA_ENABLED"] = "true"
+		Config.KumaCpConfig.Multizone.UniZone1.Envs["KUMA_EXPERIMENTAL_KDS_DELTA_ENABLED"] = "true"
+		Config.KumaCpConfig.Multizone.UniZone2.Envs["KUMA_EXPERIMENTAL_KDS_DELTA_ENABLED"] = "true"
```
Comment (michaelbeaumont) [inline test/framework/config.go:158]: Does this need to be enabled on the global cluster?

**Review by michaelbeaumont — APPROVED** [review]

File: test/framework/config.go (hunk 4 above)
Comment (lukidzi) [reply test/framework/config.go:269]: Yes, it's overridden by env

**Pull request merged**